.github/workflows/testing-and-coverage.yml

```diff
@@ -36,3 +36,26 @@ jobs:
       uses: codecov/codecov-action@v5
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
+
+  test-lowest-versions:
+    
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.9'
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        python -m pip install --upgrade uv
+        uv venv venv
+        source venv/bin/activate
+        uv pip compile --resolution=lowest -o requirements_lowest.txt pyproject.toml
+        uv pip install --constraint=requirements_lowest.txt -e '.[dev]'
```

Comment:
You don't need `'` around dev, as this isn't running on MacOS.

Reply:
Just wanted to make it copy-pastable, CI is good for examples =)

Reply:
It is actually not a macOS thing, it is zsh vs bash. I do use zsh on Linux!